addons/website_event/controllers/main.py

```diff
@@ -264,20 +235,22 @@ def _process_registration_details(self, details):
 
     @http.route(['/event/<model("event.event"):event>/registration/confirm'], type='http', auth="public", methods=['POST'], website=True)
     def registration_confirm(self, event, **post):
-        cr, uid, context = request.cr, request.uid, request.context
-        Registration = request.registry['event.registration']
+        Registration = request.env['event.registration']
         registrations = self._process_registration_details(post)
 
         registration_ids = []
+        user = request.env.user
         for registration in registrations:
-            registration['event_id'] = event
             registration_ids.append(
-                Registration.create(
-                    cr, SUPERUSER_ID,
-                    Registration._prepare_attendee_values(cr, uid, registration),
-                    context=context))
-
-        attendees = Registration.browse(cr, uid, registration_ids, context=context)
+                Registration.sudo().create({
+                    'name': registration.get('name', user.name),
+                    'phone': registration.get('phone', user.phone),
+                    'email': registration.get('email', user.email),
+                    'partner_id': user.partner_id.id,
+                    'event_id': event.id,
+                }).id)
+
+        attendees = Registration.browse(registration_ids)
```

Comment:
Instead of re-browsing can't we do something like(while creating event.registration records):
```
attendees = request.env['event.registration']
attendees += Registration.sudo().create({....})
```